src/devices/src/virtio/queue.rs

```diff
@@ -393,6 +401,64 @@ impl Queue {
         let addr = self.avail_ring.unchecked_add(2);
         Wrapping(mem.read_obj::<u16>(addr).unwrap())
     }
+
+    /// Fetch the used event (`virtq_avail->used_event`) from guest memory.
+    /// This field is used when notification suppression is enabled. It is written by the driver,
+    /// in order to specify how far the device can progress before a notification is required.
+    #[inline(always)]
+    pub fn used_event(&self, mem: &GuestMemoryMmap) -> Wrapping<u16> {
+        // We need to find the `used_event` field from the avail ring.
+        let used_event_addr = 4 + 2 * self.actual_size();
+
+        // Complete all the writes in add_used() before reading the event.
+        fence(Ordering::SeqCst);
```

Comment:
AFAIK `Ordering::Acquire` is not enough. Here we need a store-load barrier (we need to make sure that the store to `used_idx` was completed before the following load. So we need to actually stop the current CPU until all the stores are visible to all other CPUs. This can be done only with a SeqCst barrier as far as i know. `Ordering::Acquire` only provides a load-load barrier when coupled with a  `Ordering::Release` fence.